pkg/querier/frontend/frontend.go

```diff
@@ -32,19 +32,22 @@ import (
 const (
 	// StatusClientClosedRequest is the status code for when a client request cancellation of an http request
 	StatusClientClosedRequest = 499
+	defaultMaxBodySize        = 10 * 1024 * 1024 // 10 MiB
 )
 
 var (
-	errTooManyRequest   = httpgrpc.Errorf(http.StatusTooManyRequests, "too many outstanding requests")
-	errCanceled         = httpgrpc.Errorf(StatusClientClosedRequest, context.Canceled.Error())
-	errDeadlineExceeded = httpgrpc.Errorf(http.StatusGatewayTimeout, context.DeadlineExceeded.Error())
+	errTooManyRequest        = httpgrpc.Errorf(http.StatusTooManyRequests, "too many outstanding requests")
+	errCanceled              = httpgrpc.Errorf(StatusClientClosedRequest, context.Canceled.Error())
+	errDeadlineExceeded      = httpgrpc.Errorf(http.StatusGatewayTimeout, context.DeadlineExceeded.Error())
+	errRequestEntityTooLarge = httpgrpc.Errorf(http.StatusRequestEntityTooLarge, "http: request body too large")
 )
 
 // Config for a Frontend.
 type Config struct {
 	MaxOutstandingPerTenant int           `yaml:"max_outstanding_per_tenant"`
 	CompressResponses       bool          `yaml:"compress_responses"`
 	DownstreamURL           string        `yaml:"downstream_url"`
+	DownstreamMaxBodySize   int64         `yaml:"downstream_max_body_size"`
```

Comment:
In the query-frontend config, the naming "downstream" is used for the connection between query-frontend and the backend Prometheus server (or Prometheus-compatible). The max body size configurable here is not the downstream one, but it's the max input body size. For this reason I would just call it `max_body_size` (same for CLI flag, fixing flag description too).

Reply:
Changed to `MaxBodySize`